mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala

```diff
@@ -45,9 +45,12 @@ private[libsvm] class LibSVMOutputWriter(
 
   private val writer = CodecStreams.createOutputStreamWriter(context, new Path(path))
 
-  override def write(row: Row): Unit = {
-    val label = row.get(0)
-    val vector = row.get(1).asInstanceOf[Vector]
+  // This `asInstanceOf` is safe because it's guaranteed by `LibSVMFileFormat.verifySchema`
```

Comment:
`LibSVMFileFormat.verifySchema` is only called in the [`buildReader` ](https://github.com/cloud-fan/spark/blob/79bb30cf222c43c98d4d52ab207d65fdca1f83b5/mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala#L143), but this is the write path, right?